networking_aci/plugins/ml2/drivers/mech_aci/common.py

```diff
@@ -71,22 +75,121 @@ def get_address_scope_name(self, context, subnet_pool_id):
+            hg.mode = hostgroup_mode
+        return True
+
+    def get_hosts_on_segment(self, context, segment_id, level=None):
+        """Get all binding hosts (from host or binding_profile) present on a segment"""
+        # get all ports bound to segment, extract their host
+        query = context.session.query(ml2_models.PortBinding.host, ml2_models.PortBinding.profile)
+        query = query.join(ml2_models.PortBindingLevel,
+                           ml2_models.PortBinding.port_id == ml2_models.PortBindingLevel.port_id)
+        query = query.filter(ml2_models.PortBindingLevel.segment_id == segment_id)
+        if level is not None:
+            query = query.filter(ml2_models.PortBindingLevel.level == level)
+
+        hosts = set()
+        for entry in query.all():
+            host = get_host_from_profile(entry.profile, entry.host)
+            hosts.add(host)
+        return hosts
+
+    def get_hosts_on_network(self, context, network_id, level=None, with_segment=False):
+        """Get all binding hosts (from host or binding_profile) present on a network"""
+        fields = [ml2_models.PortBinding.host, ml2_models.PortBinding.profile]
+        if with_segment:
+            fields.append(ml2_models.PortBindingLevel.segment_id)
+        query = context.session.query(*fields)
+        query = query.join(ml2_models.PortBindingLevel,
+                           ml2_models.PortBinding.port_id == ml2_models.PortBindingLevel.port_id)
+        query = query.join(segment_models.NetworkSegment,
+                           ml2_models.PortBindingLevel.segment_id == segment_models.NetworkSegment.id)
+        query = query.filter(segment_models.NetworkSegment.network_id == network_id)
+        if level is not None:
+            query = query.filter(ml2_models.PortBindingLevel.level == level)
+
+        hosts = set()
+        for entry in query.all():
+            host = get_host_from_profile(entry.profile, entry.host)
+            if with_segment:
+                hosts.add((host, entry.segment_id))
+            else:
+                hosts.add(host)
+        return hosts
+
+    def get_segment_ids_by_physnet(self, context, physical_network):
+        query = context.session.query(segment_models.NetworkSegment.id)
+        query = query.filter(segment_models.NetworkSegment.physical_network == physical_network)
+        return [seg.id for seg in query.all()]
+
+    def get_network_portcount_by_physnet(self, context, physical_network):
+        query = context.session.query(segment_models.NetworkSegment.network_id,
+                                      sa.func.count(segment_models.NetworkSegment.network_id))
+        query = query.join(ml2_models.PortBindingLevel,
+                           ml2_models.PortBindingLevel.segment_id == segment_models.NetworkSegment.id)
+        query = query.group_by(segment_models.NetworkSegment.network_id)
+        query = query.filter(segment_models.NetworkSegment.physical_network == 'bb92')
```

Comment:
Why is `bb92` hardcoded here?

Reply:
Wow, that is a somewhat big oversight. This method seems to be used nowhere anyway, so I'm going to remove it.